test/object-store/sync/client_reset.cpp

```diff
@@ -237,8 +237,10 @@ TEST_CASE("sync: client reset", "[client reset]") {
         ++after_callback_invocations;
         REQUIRE(before);
         REQUIRE(before->is_frozen());
-        REQUIRE(before->read_group().get_table("class_object"));
-        REQUIRE(before->config().path == local_config.path);
+        if (!before->config().is_schema_additive()) {
```

Comment:
This is a major thing. If the schema is additive, we can't read the group without triggering an exception for the "before" realm, since the missing tables will be added, thus the OS schema will always contain a complete view of all the tables (added or deleted).

Reply:
This doesn't seem right to me - synchronized Realms will always use additive schema, so this change implies that we can't read data from the before Realm? Perhaps the correct approach here is not to update the schema of a frozen Realm, but I'll wait for @ironage weigh in.

Reply:
The failing test is:
```
-------------------------------------------------------------------------------
sync: client reset
  recovery
  Simultaneous compatible schema changes are allowed
-------------------------------------------------------------------------------
/Users/js/Documents/code/realm-core2/test/object-store/sync/client_reset.cpp:421
...............................................................................

/Users/js/Documents/code/realm-core2/test/object-store/sync/client_reset.cpp:241: FAILED:
  REQUIRE( before->read_group().get_table("class_object") )
due to unexpected exception with message:
  Unsupported schema changes were made by another client or process. For a
  synchronized Realm, this may be due to the server reverting schema changes
  which the local user did not have permission to make.
  - Property 'preexisting table name.remotely added property' has been removed.
  - Property 'same new table name.remotely added property' has been removed.
```

This test has not made any destructive schema changes, and by nature of being sync this is an additive mode. The _before_ Realm is missing properties which the server added, and this should be allowed in additive mode. This test shouldn't require the changes you made here, and I think it is pointing to a bug in your code.

Reply:
I think the problem is that I am always appending whatever the OS schema has to the list of classes (even though the new schema does not contain those classes), as per the Issue description.  The Before Realm is obviously missing those properties if they come from a class that belongs to the OS schema (and not the server one)... because the OS schema will always be a super set of the new schema we are trying to update. 

I am not sure if there is a proper solution for handling this (if not skipping the consistency check of additives schemas), but I'll keep digging, and I think the test does hit the code that I have added, since it is updating the schema.

Reply:
@ironage just pushed the minor change that allows us to get going with this new logic for additive schemas.

Reply:
this should have been fixed.